### PR TITLE
Remove \r\n from model uri path

### LINKED ISSUE #6280: [BUG] mlflow::mlflow_download_artifacts_from_uri() does not remove all relevant piece of the output
### Willingness to contribute

Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.

### MLflow version

1.27.0

### System information

- **OS Platform and Distribution**: Windows Version 10.0.19044 Build 19044
- **Python version**: 3.10.0
- **R version**: 4.1.2
- **mlflow R package version**: 1.26.1

### Describe the problem

I cannot load a model which I have previously logged.

### Tracking information

_No response_

### Code to reproduce issue

```r
mod <- carrier::crate(
  .fn = function(formula, data) stats::lm(formula = formula, data = data),
  formula, data
)

mlflow::mlflow_log_model(model = mod, artifact_path = "myCoolMod")
# 2022/07/20 14:42:35 INFO mlflow.store.artifact.cli: Logged artifact from local dir C:/.../Rtmp44ZXKk/myCoolMod to artifact_path=myCoolModRoot URI: file://C:/.../mlruns/0/bf154459eeca42a4b4f4e811ce615897/artifacts
# # A tibble: 2 x 3
#   path                is_dir file_size
#   <chr>               <lgl>      <int>
# 1 myCoolMod/MLmodel   FALSE        176
# 2 myCoolMod/crate.bin FALSE        309

mlflow::mlflow_load_model(
  model_uri = "file://C:/Users/EHYLM/work/mlruns/0/bf154459eeca42a4b4f4e811ce615897/artifacts/myCoolMod",
  flavor = "crate"
)
# Error in file(file, "rt", encoding = fileEncoding) : 
#   cannot open the connection
# In addition: Warning message:
# In file(file, "rt", encoding = fileEncoding) :
#   cannot open file 'C:/.../mlruns/0/bf154459eeca42a4b4f4e811ce615897//MLmodel': Invalid argument
```

I took a look under the hood and there are a couple of things going on here. Firstly, [`mlflow:::mlflow_download_artifacts_from_uri()`](https://github.com/mlflow/mlflow/blob/master/mlflow/R/mlflow/R/tracking-runs.R#L427-L431) is called and returns 

```
model_path
# [1] "C:\\...\\mlruns\\0\\bf154459eeca42a4b4f4e811ce615897\\artifacts\\myCoolMod\r"
```

As you can see, there has been a `\r` appended here. If you step into this function, there is a `\n` too but this is removed [here](https://github.com/mlflow/mlflow/blob/master/mlflow/R/mlflow/R/tracking-runs.R#L427-L430). I _think_ this is a bug.

If we carry on stepping through the `mlflow::mlflow_load_model()`, the following is called ([here](https://github.com/mlflow/mlflow/blob/master/mlflow/R/mlflow/R/model.R#L80))

```r
fs::path(model_path, "MLmodel")
# /MLmodel/.../work/mlruns/0/bf154459eeca42a4b4f4e811ce615897/artifacts/myCoolMod
```

As you can see, rather than appending the `MLmodel`, it has instead prepended it. This may well be a bug within the `fs` package.

### Other info / logs

_No response_

### What component(s) does this bug affect?

- [X] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [X] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [X] `area/windows`: Windows support

### What language(s) does this bug affect?

- [X] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### PR DESCRIPTION
**Pull request by nathaneastwood**
## Related Issues/PRs

Fixes #6280 

## What changes are proposed in this pull request?

This removed the "\r" returned to stdout by `mlflow:::mlflow_cli()`.

## How is this patch tested?

Locally only.

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [x] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [x] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [x] `area/windows`: Windows support

Language
- [x] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by nathaneastwood** [comment]
I haven't added any tests as I'm travelling so it's a little tricky but I think this should probably have one.

**Comment by dbczumar** [comment]
Hi @nathaneastwood , any updates here?

File: mlflow/R/mlflow/R/tracking-runs.R [hunk 1]
```diff
@@ -427,7 +427,7 @@ mlflow_download_artifacts <- function(path, run_id = NULL, client = NULL) {
 # ' Download Artifacts from URI.
 mlflow_download_artifacts_from_uri <- function(artifact_uri, client = mlflow_client()) {
   result <- mlflow_cli("artifacts", "download", "-u", artifact_uri, echo = FALSE, client = client)
-  gsub("\n", "", result$stdout)
+  gsub("[\r\n]", "", result$stdout)
```
Comment (harupy) [inline mlflow/R/mlflow/R/tracking-runs.R:430]: Can we use `trimws()` here?

https://stat.ethz.ch/R-manual/R-patched/library/base/html/trimws.html

Comment (nathaneastwood) [reply mlflow/R/mlflow/R/tracking-runs.R:430]: Yes, good shout. Looks like the correction here `[\r\n]` fixes the broken tests. So `trimws()` should also do the same job. I'll push a change now.

**Comment by nathaneastwood** [comment]
@harupy this should be good to merge now I think

**Review by harupy — APPROVED** [review]
LGTM! The windows check failure is unrelated.

**Pull request merged**